lib/format_parser.rb

```diff
@@ -14,6 +18,11 @@ def self.register_parser_constructor(object_responding_to_new)
     PARSER_MUX.synchronize do
       @parsers ||= []
       @parsers << object_responding_to_new
+      # Gathering natures from parsers.
```

Comment:
See below. We are indeed achieving a speedup by having parser natures cached here, but unless I see strong indications that this speedup is extremely significant compared to the time it takes to actually parse things this looks like overcomplicating things - instead of just asking an already instantiated parser "will you be emitting images?" and discarding it if we don't need it. There is GC overhead to everything but in this case it should not be that big of an issue, at least initially.

Reply:
In my design, we need to gather all the natures from parsers to provide default parsing options. The whole motto of this PR is to, like you said, trust that the user will make the right choices. Therefore, in case of our user will not provide any _indication_ (😅), it's assumed that she rather have all the results available. The code as-is, to not to filter out any, requires knowing what are the default values. Because of the whole mutex thingy, I thought to make it extensible rather than using a hardcoded array.

However, in case you don't like it, we can use a magic constant (`:*`) to avoid this gathering, but that's gonna make the code more brittle imho.